bundles/admin/admin-layereditor/instance.js

```diff
@@ -345,15 +346,16 @@ Oskari.clazz.defineES('Oskari.admin.admin-layereditor.instance',
                     },
                     success: function (response) {
                         setLoading(false);
-                        setValue(response.name);
+                        !newSubgroup && setValue(response.name);
+                        parentId = response.parentId;
                     }
                 });
             };
 
-            const loc = id ? this.loc('editTheme') : this.loc('addTheme');
+            const loc = newSubgroup ? this.loc('addTheme') : id ? this.loc('editTheme') : this.loc('addTheme');
             this.themeFlyout = new LocalizingFlyout(this, loc, {
                 headerMessageKey: 'themeName',
-                id: id,
+                id: id || newSubgroup && id,
```

Comment:
How does this work? I read it as id is either id if it's truthy OR id(?) if newSubgroup is truthy resulting in id = id whatever the case. If it does something different than that you should add some parenthesis to make it easier to read.

Reply:
When user will edit theme or add new sub theme, data needs to be fetched. This will basically just check if id needs to be set or not. When new root theme is added, id is not set at all. But yes this could indeed look better. I will fix it.